tasks.py

```diff
@@ -300,6 +361,170 @@ def post_upgrade(context):
+    command = [
+        "exec",
+        "--",
+        service,
+        command,
+        f"< '{file}'" if file else "",
+    ]
+    docker_compose(context, " ".join(command), pty=not bool(file))
+
+
+@task(
+    help={
+        "db-name": "Database name (default: Nautobot database)",
+        "input-file": "SQL file to execute and quit (default: empty, start interactive CLI)",
+        "output-file": "Ouput file, overwrite if exists (default: empty, output to stdout)",
+        "query": "SQL command to execute and quit (default: empty)",
+    }
+)
+def dbshell(context, db_name="", input_file="", output_file="", query=""):
+    """Start database CLI inside the running `db` container.
+
+    Doesn't use `nautobot-server dbshell`, using started `db` service container only.
+    """
+    if input_file and query:
+        raise ValueError("Cannot specify both, `input_file` and `query` arguments")
+    if output_file and not (input_file or query):
+        raise ValueError("`output_file` argument requires `input_file` or `query` argument")
+
+    env = {}
+    if query:
+        env["_SQL_QUERY"] = query
+
+    command = [
+        "exec",
+        "--env=_SQL_QUERY" if query else "",
+        "-- db sh -c '",
+    ]
+
+    if _is_compose_included(context, "mysql"):
+        command += [
+            "mysql",
+            "--user=$MYSQL_USER",
+            "--password=$MYSQL_PASSWORD",
+            f"--database={db_name or '$MYSQL_DATABASE'}",
+        ]
+    elif _is_compose_included(context, "postgres"):
+        command += [
+            "psql",
+            "--username=$POSTGRES_USER",
+            f"--dbname={db_name or '$POSTGRES_DB'}",
+        ]
+    else:
+        raise ValueError("Unsupported database backend.")
+
+    command += [
+        "'",
+        '<<<"$_SQL_QUERY"' if query else "",
+        f"< '{input_file}'" if input_file else "",
+        f"> '{output_file}'" if output_file else "",
+    ]
+
+    docker_compose(context, " ".join(command), env=env, pty=not (input_file or output_file or query))
+
+
+@task(
+    help={
+        "input-file": "SQL dump file to replace the existing database with. This can be generated using `invoke backup-db` (default: `dump.sql`).",
+    }
+)
+def import_db(context, input_file="dump.sql"):
+    """Stop Nautobot containers and replace the current database with the dump into the running `db` container."""
+    docker_compose(context, "stop -- nautobot worker beat")
+    start(context, "db")
+    _await_healthy_service(context, "db")
+
+    command = ["exec -- db sh -c '"]
+
+    if _is_compose_included(context, "mysql"):
+        command += [
+            "mysql",
+            "--database=$MYSQL_DATABASE",
+            "--user=$MYSQL_USER",
+            "--password=$MYSQL_PASSWORD",
+        ]
+    elif _is_compose_included(context, "postgres"):
+        command += [
+            "psql",
+            "--username=$POSTGRES_USER",
+            "postgres",
```

Comment:
```suggestion
            "$POSTGRES_DB",
```

Reply:
Update import-db postgres db name to use env var. Ref: https://github.com/nautobot/cookiecutter-nautobot-app/issues/38

Reply:
Target DB name to be imported is defined inside the dump file. `postgres` here, is just the default DB name, to connect to, to be able to `DROP` and `CREATE` the target DB.

Reply:
Potentially we can add `invoke import-db --drop-db --create-db` arguments for cases, when these commands are not include in the dump file.

Reply:
Question: Do we want to be able to import additional SQL data into the existing database without flushing it?

cc.: @whitej6

Reply:
No to additional DBs. Only support will be destroy container + volume, the recreate container and populate single DB for nautobot.

Reply:
@snaselj shouldn't we just keep the default DB name to be from `$POSTGRES_DB` instead of `postgres`? Changing it to `$POSTGRES_DB` is what fixed the issue for me (see https://github.com/nautobot/cookiecutter-nautobot-app/issues/38)

And yes, we'll want to make sure the task can complete even if the `DROP` command isn't in the `dump.sql` like it wasn't for the dataset I'm trying to import.

Reply:
The current `invoke backup-db` implementation adds `DROP DATABASE` and `CREATE DATABASE` statements to the backup file. `invoke import-db`, as it's counterpart, doesn't need to handle database creation, as it's already done, when running SQL backup file. For this to work, it's necessary for `psql` client, to connect to the default database `postgres`, to be able to drop and create `nautobot` (or other) database.

However, this way, it's not compatible with dump files not containing these flush statements. That's why I'm suggesting adding `--drop-and-create=<dbname>`. This argument should be used in cases, where `DROP DATABASE` and `CREATE DATABASE` are not included in SQL dump file.

I'm opposed to remove docker volume, this way the task will be dependent on the docker volume name, which is an implementation detail. It's always better to avoid accessing hidden implementation details like this, even thought it's possible.

Reply:
Adding `--drop-and-create=<dbname>` should be good then. Just wanted to make sure there's a way to import that doesn't require using the `invoke backup-db` task to generate the dump.

Reply:
Reconsidered the output from `invoke backup-db`. `dump.sql` now doesn't contain flush statements and the database name, to be able to implement `invoke import-db --db-name <database name>` correctly.

Added `invoke destroy-and-import-db` alternative solution.

Tested with both, MySQL and PostgreSQL.

cc.: @whitej6

Reply:
So the main difference between `invoke import-db` and `invoke destroy-and-import-db` is the first keeps the existing volumes and just replaces the one that's used in Nautobot? And the second removes all the existing volumes?

Reply:
Correct.